File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/db/user/UsersPersistanceDBModule.java
package com.fintechplatform.ui.profile.db.user;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class UsersPersistanceDBModule {

    @Provides
    @Singleton
    UsersPersistanceDB providesUsersPersistanceDB(UsersDB usersDB) {
        return new UsersPersistanceDB(usersDB);
    }

    @Provides
    @Singleton
    UsersDB providesUserDB() {
        return new UsersDB();
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/db/user/UsersPersistanceDBModule.kt
package com.fintechplatform.ui.profile.db.user

import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class UsersPersistanceDBModule {

    @Provides
    @Singleton
    internal fun providesUsersPersistanceDB(usersDB: UsersDB): UsersPersistanceDB {
        return UsersPersistanceDB(usersDB)
    }

    @Provides
    @Singleton
    internal fun providesUserDB(): UsersDB {
        return UsersDB()
    }
}
